jupyter_book_to_htmlbook/chapter_processing.py

```diff
@@ -231,8 +231,22 @@ def process_chapter(toc_element, build_dir=Path('.')):
     chapter = process_math(chapter)
     chapter = move_span_ids_to_sections(chapter)
 
-    # write the file
-    out = build_dir / (ch_name + '.html')
+    # write the file, preserving any directory structure(s) from source
+    if type(toc_element) == list:
+        dir_structure = [p for p in toc_element[0].parts
+                         if p not in source_dir.parts]
+    else:
+        dir_structure = [p for p in toc_element.parts
+                         if p not in source_dir.parts]
+    parents = '/'.join(dir_structure[:-1])  # don't double the file stem
+    if parents:
```

Comment:
All this "parents" stuff is to get around the possibility of someone using a /ch1/chapter.ipynb /ch2/chapter.ipynb kind of structure, where before the second resultant 'chapter.html' would have rewritten the first, now they will appear as /ch1/chapter.html and /ch2/chapter.html in the target/build folder (maybe as expected, or at least: as a way to keep everything separate).